### PR TITLE
Require chemistry tools to craft mutagens

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Balance "Chemistry tool qualities are now required to craft mutagens"```

#### Purpose of change
I added chemistry tool qualities and chemistry tools to the game some time ago. They need implementation to various recipes, which in turn I hope will encourage people with chemistry knowledge to add their own favourite reactions.

Mutagens are a late-game play item that require a lot of balancing. I have written at length about the major things that I think should be done: https://github.com/CleverRaven/Cataclysm-DDA/issues/28277. In the shorter term though, making mutagens a little less accessible early on would be nice.

#### Describe the solution
One tiny drop in the bucket of mutagen balancing is to require better chemistry tools to make them: currently they are too easily accessed, this adds a small amount more time gating to full mutagen accessibility. Further, this provides a demonstration of the common tool qualities for chemistry and gives these items an important and realistic use.

Also I made mutagens take longer to craft. Fifteen minutes? Really? Regular mutagens gain a lot from batching, because you can do a single separation and analysis run on a large amount. Serums don't gain as much because it takes longer to distill and concentrate a large batch.

Finally, while I was in there, I took out some of the books that no longer make lore sense to contain mutagen recipes.

#### Describe alternatives you've considered
I would consider making IV mutagens require a spectrophotometer and use charges.

#### Additional context
I created a new file for mutagen recipes since they're a discrete item with a lot of entries, they don't need to clutter up medandchemicals.json

I expect this will win me a lot of fans and admirers.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
> I expect this will win me a lot of fans and admirers.

Too bad adding additional gates usually adds only haters.

**Comment by I-am-Erk** [comment]
I'm sure there will be a strong contingent of people who are excited to get to use a separation funnel in game! Certain of it!

File: data/json/recipes/chem/mutagens.json [hunk 1]
```diff
@@ -0,0 +1,790 @@
...
+    "difficulty": 9,
+    "time": "45 m",
+    "batch_time_factors": [ 80, 20 ],
+    "book_learn": [ [ "recipe_animal", 8 ] ],
+    "qualities": [ { "id": "CHEM", "level": 3 }, { "id": "SEPARATE", "level": 1 }, { "id": "ANALYSIS", "level": 1 } ],
+    "tools": [ [ [ "surface_heat", 25, "LIST" ] ] ],
+    "components": [
+      [ [ "mutagen", 1 ] ],
+      [
+        [ "cheese", 1 ],
```
Comment (kevingranade) [inline data/json/recipes/chem/mutagens.json]: Alchemy.
Needs more liquified mice.

Comment (I-am-Erk) [reply data/json/recipes/chem/mutagens.json]: That could be arranged.

**Comment by Rail-Runner** [comment]
In case anyone's interested to see possible alternatives: I considered doing something similar at one point, my plan was to get those mutagen recipes to use some kind of semi-portable (you can haul it around, but it's not as compact as a lighter) "bioengineering suitcase lab" that can only be found in various places, but not crafted. 

It could take genetic material from samples (used as tools, because amount taken is negligible), then use a built-in irradiator and/or some other things that use up power to irradiate it and the blob globs together to produce "targeted" mutagen (untargeted mutagen doesn't need any samples; just blob globs). Samples aren't abstracted away, though; for instance, to make "mouse mutagen", one would actually need to go and catch a mouse somewhere to get a chunk of its flesh. Some of the more rare mutagens related to species that don't exist in the wild (like raptor mutagen) would have to use samples that can only be found in labs, more "chaotic" ones (like chimera mutagen) would need just a chunk of flesh, the origin of which doesn't matter. In addition, if this doesn't feel balanced as it is, the recipe could use some sort of common/rare craftable/uncraftable chemicals to influence the blob by causing specific "hostile environment" and forcing blob to adapt to it in a specific way, not just irradiation.

Samples do spoil normally, but could be quite easily preserved indefinitely in a jar with ethanol (and be found in this state in labs).

The suitcase lab could also be used for other biological projects later (once someone would make related recipes), like culturing microorganisms or making vaccines, but for those purposes, a set of related "normal" gear would be enough as well.

File: data/json/recipes/chem/mutagens.json [hunk 2]
```diff
@@ -0,0 +1,777 @@
...
+    "skill_used": "cooking",
+    "skills_required": [ "firstaid", 1 ],
+    "difficulty": 9,
+    "time": "45 m",
+    "batch_time_factors": [ 80, 20 ],
+    "book_learn": [ [ "recipe_animal", 8 ] ],
+    "qualities": [ { "id": "CHEM", "level": 3 }, { "id": "SEPARATE", "level": 1 }, { "id": "ANALYSIS", "level": 1 } ],
+    "tools": [ [ [ "surface_heat", 25, "LIST" ] ] ],
+    "components": [ [ [ "mutagen", 1 ] ], [ [ "meat", 1 ] ], [ [ "ammonia", 1 ], [ "lye_powder", 100 ] ] ],
+    "//": "fun fact: mice don't like cheese as much as sugar! we include it in the recipe because most people correlate the two, and as a fun poke at it"
```
Comment (anothersimulacrum) [inline data/json/recipes/chem/mutagens.json]: This comment is irrelevant now.
```suggestion
```

**Comment by Fris0uman** [comment]
Do we want to change the recipe at some point? Like changing `honey` and `wax` for actual insect meat for example?

**Comment by Amoebka** [comment]
The change is flavourful, but I'm not sure what exactly it achieves gameplay-wise. Chemical tools are found in the same place mutagen recipes are - underground labs, so they will be acquired at roughly the same time as before. I would also like to point out that gating mutations behind tools does little to adress the "all-or-nothing" issue with them. Since the **ingredients** are dirt common, once the player acquires the recipes (and now also tools), they can instantly go from "no mutations" to "full post-threshold in the desired category".

>Do we want to change the recipe at some point? 

I think this is the only option. And, preferably, the ingredients should be a bottleneck in themselves, not trivially obtainable items like meats. Maybe mutagens can be split into "low-level" that are craftable from common things (i.e. insect meat), while going post-threshold requires rare materials (i.e. meat from ant queens specifically).

**Comment by Zireael07** [comment]
@Amoebka: note that item spawn is random, it's the same deal as  "welders in garages" - you might find 2 in one place, you might visit 10 places without finding one. And requiring chemistry tools makes perfect sense from both logic and gameplay point of view.

**Comment by Amoebka** [comment]
> note that item spawn is random

Same can be said about mutagen recipe books. Generally, if a character is strong enough to explore one lab to find a recipe book they are strong enough to explore however many labs it takes to acquire all the tools. As such, the change doesn't really push mutations later into the game.

I'm not against the change in any way, I'm just pointing out that it by itself achieves a lot less that might seem initially, and follow up changes (preferably gating the mutagen ingredients) are necessary before any improvements can be claimed.

**Comment by Muffindrake** [comment]
> but I'm not sure what exactly it achieves gameplay-wise

It does precisely nothing, these changes only add time gating for the sake of it. It's not clear how mutagen is supposed to be reworked, because despite it currently being a very broken mechanic, it's at least somewhat balanced by the fact that you have to loot a dangerous lategame facility for them. 

All this does now is effectively gut Cook starts to oblivion where you have the minimum points required to start with all the recipes you want, and everyone else suffers on top of that. Mutations need to be less of an automatic exclusive stat/power up and more on the double-edged sword side of things in order to get things moving into a more reasonable direction. Adding 17 hours to the baking time doesn't fix your brownie charcoal.

**Comment by Mecares** [comment]
To be fair cooks with a campfire and a chemistry set being able to do the work that should only be able to be done by some kind of genius bioscientist with a modern lab with the right equipment and research materials, is broken itself and not really something i consider important to protect. Getting access to endgame content on day 1 is not desireable from a game balance point even if some sacrifices are made during char generation in my humble opinion.

**Comment by Fris0uman** [comment]
>Cook Start
But making mutagen was never available from the start. It has already been pointed out that the tool required for this are in the same place as the recipe ( plus above ground lab I suppose) so it doesn't gate that much. But it does make more sens and it lays the groundwork for more sensible mutagen recipes.

**Comment by I-am-Erk** [comment]
>One tiny drop in the bucket

I'm not labouring under the illusion that this fixes mutagens. It is, however, a necessary change, and it does add a very small time gate. There's much, much more in the pipeline but this is a nice simple place to start.

**Comment by kevingranade** [comment]
@Muffindrake can you expand on what you mean by cook starts? If a scenario has the player starting with access to mutagen recipes, that's a *bug* and needs to be fixed.

**Comment by Mecares** [comment]
If i remember correctly there is something bugged with auto learning recipes on character generation which ignores book only recipes and makes you auto learn all recipes you have the required skills for.

**Comment by I-am-Erk** [comment]
Well, that's a whole other set of problems. If this helps bandaid people starting the game with top-secret mutagen serums, that's not a negative.

**Comment by anothersimulacrum** [comment]
> If i remember correctly there is something bugged with auto learning recipes on character generation which ignores book only recipes and makes you auto learn all recipes you have the required skills for.

That's not a bug, it's intentional #14102.

**Comment by I-am-Erk** [comment]
Added intentionally perhaps, but:

>If a scenario has the player starting with access to mutagen recipes, that's a bug and needs to be fixed.

I'd describe it as an unintended consequence of 14102.

**Comment by Muffindrake** [comment]
> @Muffindrake can you expand on what you mean by cook starts? If a scenario has the player starting with access to mutagen recipes, that's a _bug_ and needs to be fixed.

It's not different from any other starts, just that Cook starts with a base 4 in Cooking, so you need only the minimal amount of points to get to Cooking 10 at character generation, which due to recipe system design starts you out with all mutagen recipes available at start, and all serums if you also take first aid 3.

**Pull request merged**